### DISCUSSION TITLE
Removal of Highlights

### DISCUSSION THREAD

**Discussion by lunamaze** [pymupdf/PyMuPDF#3262] [Looking for help] [answered]
Hello all. I would like to remove the highlights in a PDF. These highlights seem to be vector graphics rather than annotations, as `page.first_annot` returns empty list but `page.get_cdrawings()` returns a list that contains the highlight objects (identified by colour code in 'fill'). 

I have searched and found in #2308 that removal of shapes are currently not supported. Are there any methods that I can adopt to remove the highlights (while keeping the text not affected)?

**Comment by JorjMcKie** [comment, marked as answer]
Removal of line-art / vector graphics will be supported soon (in the next version).
But you are requesting to let survive text when applying redactions - and this remains **_not_** supported.

Interestingly, we have been discussing this type of flexibility just recently and are considering this as a possible next feature. But the decision to do so has not been made yet.
